src/stores/repository/RemoteRepository.ts

```
@@ -34,9 +34,20 @@ export default abstract class RemoteRepository<Item, Collection> {
    * @example
    * const notifications = await repo.findBy({ unread: true });
    */
-  async findBy(queryParams: any): Promise<Collection> {
-    const json = await fetchAPI(this.remotePathOrUrl, queryParams);
-    return humps.camelizeKeys(json);
+  async findBy(queryParams: any): Promise<Collection | void> {
+    try {
+      const json = await fetchAPI(this.remotePathOrUrl, queryParams);
+      return humps.camelizeKeys(json);
+    } catch (error) {
+      // Don't throw Network Errors. Browsers log these themselves and the realtime connection makes
+      // MagicBell sensitive to connection errors. We don't want to have these flooding consumer logs.
+      if (/Network Error/.test(error.message)) {
+        return;
```

Note that I'm completely silencing the error. I thought it made sense to log it, but the browser already logs them anyway. So I don't think it's necessary to log it explicitly as well.

One use case to log regardless would be for log intercepting. (`console.log = (...args) => myLogger.log(...args)`). Let's see how this works. We can always add a log statement if the need arises.